### PR TITLE
Broke Pulsar-IO::HDFS into 3.x and 2.x versions

### PR DESCRIPTION
**Pull request by david-streamlio**
### Motivation

A user was attempting to use the existing HDFS connector to connect to a 2.x version of HDFS, but the current connector only supported 3.x version of HDFS. 

### Modifications

To address this issue, we renamed the current HDFS connector to HDFS3, and created a new 2.x compatible connector named HDFS2. 

The code in both of these are nearly identical with 2 notable exceptions. First and foremost, they both use different versions of the Hadoop-client library. And secondly, the HDFS2 version creates the FSDataOutputStream object directly, whereas the HDFS3 version leverages the FSDataOutputStreamBuilder class for this purpose, as it is the preferred method going forward.

### Result

There will be be support for connecting to both 2.x and 3.x version of HDFS. However, there MAY BE some library conflicts in the released jar due to the different versions of the same library in the different modules. Hopefully the NAR packaging will address this.

### CODE REVIEW

File: pulsar-io/hdfs3/src/test/java/org/apache/pulsar/io/hdfs3/sink/text/HdfsStringSinkTests.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+public class HdfsStringSinkTests extends AbstractHdfsSinkTest<String, String> {
+	
+    @Override
+    protected void createSink() {
+        sink = new HdfsStringSink();
+    }
+    
+    @Test(enabled = false)
+	public final void write5000Test() throws Exception {
```
Comment (sijie) [inline pulsar-io/hdfs3/src/test/java/org/apache/pulsar/io/hdfs3/sink/text/HdfsStringSinkTests.java]: there seems to be indent issues in this file?

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by jiazhai** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Pull request merged**